pkg/controller/localvolumeset/finalizer.go

```diff
@@ -0,0 +1,60 @@
+package localvolumeset
+
+import (
+	"context"
+	"fmt"
+
+	localv1alpha1 "github.com/openshift/local-storage-operator/pkg/apis/local/v1alpha1"
+	"github.com/openshift/local-storage-operator/pkg/common"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *LocalVolumeSetReconciler) syncFinalizer(request reconcile.Request) error {
+	lvSet := &localv1alpha1.LocalVolumeSet{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, lvSet)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			r.lvSetMap.deregisterLocalVolumeSet(lvSet.Spec.StorageClassName, request.NamespacedName)
+			return nil
+		}
+		return fmt.Errorf("failed to get localvolumeset: %w", err)
+	}
+
+	setFinalizer := false
+	changed := false
+
+	if lvSet.DeletionTimestamp.IsZero() {
+		// finalizer should exist and be removed only when deleting
+		setFinalizer = true
+
+	} else {
+		// fetch PVs that match the storageclass
+		pvList := &corev1.PersistentVolumeList{}
+		err = r.client.List(context.TODO(), pvList, client.MatchingFields{pvStorageClassField: lvSet.Spec.StorageClassName})
```

Comment:
Using storageclass name in PV is not a good way of tracking PVs that were created by LSO.  storageclass can be deleted or can exist before localvolumset object was created. Or how about - I create 2 localvolumset objects with same SC name, now I can't delete one of them if other one is in use..

Reply:
> I create 2 localvolumset objects with same SC name, now I can't delete one of them if other one is in use..

Since I don't think it'll be possible to associate a PV with a localvolumeset beyond the one-to-many mapping of the storagecluster relationship, I believe we will have to make sure there are no bound PVs associated with a storageclass before the corresponding lvsets are allowed to be removed as you say.

Reply:
I guess this depends on the provisioner-as-a-library change

Reply:
It is possible to specify PV labels in configmap - https://github.com/openshift/local-storage-operator/blob/master/pkg/controller/localvolume/localvolume_controller.go#L803 which we are doing for localvolume objects. But in this case it is tricky because we are creating one configmap for all localvolumeset objects. I would settle for a label that tracks all of these PVs that they were created by `localvolumeset` object rather than waiting for provisioner-as-library change.

Reply:
To identify objects created by a particular `localvolumeset` object or any `localvolumeset`? The latter can be done easily, but the former can't be done without a different configmap/provisioner-daemonset per `localvolumeset`

Reply:
hmm yeah, there is no good way to track PVs created for particular `localvolumeset` object. This is however a problem - and I don't like to document yet another workaround.

Reply:
Have updated based on new LocalVolumeSet ownership labels